dts/bindings: document use of value 'use-prop-name'

erwango opened the pull request:
'use-prop-name' is not documented.
Update dts/bindings/device_node.yaml.template to fix this.

Fixes #9971

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

SebastianBoe commented on dts/bindings/device_node.yaml.template:
```diff
@@ -61,4 +61,9 @@ properties:
   - cell2    # name of third cell
   - and so on and so forth
+
+# In case "generation" category is set to "define", value "use-prop-name"
```
The word 'category' is already used and means something else, I hope we can use a different word.

SebastianBoe replied:
Attribute maybe?

erwango replied:
ok, corrected.

SebastianBoe commented on dts/bindings/device_node.yaml.template:
```diff
@@ -61,4 +61,9 @@ properties:
   - cell2    # name of third cell
   - and so on and so forth
+
+# In case "generation" category is set to "define", value "use-prop-name"
```
s/In case/When the/

erwango replied:
Done.

erwango commented:
@SebastianBoe: changes done. Thanks for the review!

erwango committed:
dts/bindings: document use of value 'use-prop-name'

'use-prop-name' is not documented.
Update dts/bindings/device_node.yaml.template to fix this.
Document that 'type' attribute is not used.

Fixes #9971

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

dbkinder reviewed (approved):
OK for me, no .rst files updated though. Just the template.
